orderBook.js

```diff
@@ -0,0 +1,45 @@
+
+
+const isMatchingOrder = (existingBook, incomingOrder) => {
+  if (existingBook.type !== incomingOrder.type) {
+    if (incomingOrder.type === 'sell') {
+      if (existingBook.price >= incomingOrder.price) {
+        return true
+      }
+    } else if (existingBook.price <= incomingOrder.price) {
+      return true
+    }
+  }
+  return false
+}
+
+function reconcileOrder(existingBook, incomingOrder) {
+
+  let updatedBook = [...existingBook]
```

Comment:
If you started with an empty array and filled it as you went you wouldn't have needed to change the order in the test